runtime/compilers/rillv1/parser.go

```diff
@@ -184,6 +184,9 @@ type Parser struct {
 	Resources map[ResourceName]*Resource
 	Errors    []*runtimev1.ParseError
 
+	// Mapping of local data files to resource paths that depend on them
+	localDataToResourcepath map[string]map[string]any
+
 	// Internal state
 	resourcesForPath           map[string][]*Resource // Reverse index of Resource.Paths
 	resourcesForUnspecifiedRef map[string][]*Resource // Reverse index of Resource.rawRefs where kind=ResourceKindUnspecified
```

Comment:
Since the parser already keeps state of the paths for a particular `ResourceName` (through `Parser.Resources[name]`), could it just store the resource names that use a local data file?

Also, can it put it under the `// Internal state` section? Ordering of fields indicate their importance and relationship, and this is for example a less central piece of state than `resourcesForPath`.

 For example:

```suggestion
	// Internal state
	resourcesForPath           map[string][]*Resource // Reverse index of Resource.Paths
	resourcesForUnspecifiedRef map[string][]*Resource // Reverse index of Resource.rawRefs where kind=ResourceKindUnspecified
	resourceNamesForDataPaths map[string][]ResourceName // Index of local data files to resources that depend on them
```